src/components/EmojiPicker/index.js

```diff
@@ -74,39 +41,41 @@ class EmojiPicker extends React.Component {
         this.emojiPopoverDimensionListener.remove();
     }
 
-    /**
-     * Update preferredSkinTone and sync with Onyx, NVP.
-     * @param {Number|String} skinTone
-     */
-    setPreferredSkinTone(skinTone) {
-        if (skinTone === this.props.preferredSkinTone) {
-            return;
-        }
-
-        User.setPreferredSkinTone(skinTone);
-    }
-
     /**
      * Callback for the emoji picker to add whatever emoji is chosen into the main input
      *
      * @param {String} emoji
-     * @param {Object} emojiObject
      */
-    selectEmoji(emoji, emojiObject) {
-        EmojiUtils.addToFrequentlyUsedEmojis(this.props.frequentlyUsedEmojis, emojiObject);
+    selectEmoji(emoji) {
         this.hideEmojiPicker();
-        this.props.onEmojiSelected(emoji);
+        if (_.isFunction(this.onEmojiSelected)) {
+            this.onEmojiSelected(emoji);
+        }
     }
 
     hideEmojiPicker() {
         this.setState({isEmojiPickerVisible: false});
     }
 
-    showEmojiPicker() {
-        if (_.isFunction(this.props.onBeforeShowEmojiPicker)) {
-            this.props.onBeforeShowEmojiPicker();
+    /**
+     * Show the ReportActionContextMenu modal popover.
+     *
+     * @param {Function} [onModalHide=() => {}] - Run a callback when Modal hides.
+     * @param {Function} [onEmojiSelected=() => {}] - Run a callback when Emoji selected.
+     * @param {Function} [onBeforeShowEmojiPicker=() => {}] - Run a callback before showing EmojiPicker
+     * @param {Element} emojiPopoverAnchor - Element to which Popover is anchored
+     */
+    showEmojiPicker(onModalHide, onEmojiSelected, onBeforeShowEmojiPicker, emojiPopoverAnchor) {
+        if (_.isFunction(onBeforeShowEmojiPicker)) {
+            onBeforeShowEmojiPicker();
         }
```

Comment:
This indicates that this should have never been passed as a param. Just call it when `showEmojiPicker` is called.

Reply:
This one won't be required and I've removed it.